engine/annotations/depth.py

```diff
@@ -0,0 +1,71 @@
+import bpy
+import gpu
+from gpu_extras.batch import batch_for_shader
+import numpy as np
+import threading
+
+def render_depth_map(context, collection=None, invert=True, width=None, height=None, matrix=None, projection_matrix=None, main_thread=False):
+    e = threading.Event()
+    result = None
+
+    width, height = width or context.scene.render.resolution_x, height or context.scene.render.resolution_y
+    matrix = matrix or context.scene.camera.matrix_world.inverted()
+    projection_matrix = projection_matrix or context.scene.camera.calc_matrix_camera(
+        context,
+        x=width,
+        y=height
+    )
+
+    def _execute():
+        nonlocal result
+        offscreen = gpu.types.GPUOffScreen(width, height)
+
+        with offscreen.bind():
+            fb = gpu.state.active_framebuffer_get()
+            fb.clear(color=(0.0, 0.0, 0.0, 0.0), depth=1)
+            gpu.state.depth_test_set('LESS_EQUAL')
+            gpu.state.depth_mask_set(True)
+            with gpu.matrix.push_pop():
+                gpu.matrix.load_matrix(matrix)
+                gpu.matrix.load_projection_matrix(projection_matrix)
+                
+                shader = gpu.shader.from_builtin('3D_UNIFORM_COLOR')
+
+                for object in (context.scene.objects if collection is None else collection.objects):
+                    object = object.evaluated_get(context)
+                    try:
+                        mesh = object.to_mesh(depsgraph=context)
+                    except:
+                        continue
+                    if mesh is None:
+                        continue
+                    mesh.transform(object.matrix_world)
+                    mesh.calc_loop_triangles()
+                    vertices = np.empty((len(mesh.vertices), 3), 'f')
+                    indices = np.empty((len(mesh.loop_triangles), 3), 'i')
+
+                    mesh.vertices.foreach_get("co", np.reshape(vertices, len(mesh.vertices) * 3))
+                    mesh.loop_triangles.foreach_get("vertices", np.reshape(indices, len(mesh.loop_triangles) * 3))
+                    
+                    batch = batch_for_shader(
+                        shader, 'TRIS',
+                        {"pos": vertices},
+                        indices=indices,
+                    )
+                    batch.draw(shader)
+            depth = np.array(fb.read_depth(0, 0, width, height).to_list())
+            if invert:
+                depth = 1 - depth
+                mask = np.array(fb.read_color(0, 0, width, height, 4, 0, 'UBYTE').to_list())[:, :, 3]
+                depth *= mask
+            depth = np.interp(depth, [np.ma.masked_equal(depth, 0, copy=False).min(), depth.max()], [0, 1]).clip(0, 1)
+        offscreen.free()
```

Comment:
prevent render window from flickering
```suggestion
        gpu.state.depth_test_set('NONE')
        offscreen.free()
```